test/integration/CalldataExecute.t.sol

```diff
@@ -28,8 +28,126 @@ contract CalldataExecute is Test, Configs {
     }
 
     function testSimulateExec() public {
-        vm.warp(block.timestamp + 24 hours);
-        TemporalGovernor(addresses.getAddress("TEMPORAL_GOVERNOR"))
-            .executeProposal(execData);
+        address caller = address(this);
+        address governanceToken = addresses.getAddress("WELL");
+        address governorAddress = addresses.getAddress(
+            "MULTICHAIN_GOVERNOR_PROXY"
+        );
+        address stellaSwapRewarder = addresses.getAddress(
+            "STELLASWAP_REWARDER"
+        );
+        MultichainGovernor governor = MultichainGovernor(governorAddress);
+
+        vm.prank(0x6972f25AB3FC425EaF719721f0EBD1Cdb58eE451);
+        ERC20Votes(governanceToken).approve(governorAddress, type(uint256).max);
+
+        deal(governanceToken, governorAddress, 100_000_000 * 1e18);
+
+        {
+            // Ensure proposer has meets minimum proposal threshold and quorum votes to pass the proposal
+            uint256 quorumVotes = governor.quorum();
+            uint256 proposalThreshold = governor.proposalThreshold();
+            uint256 votingPower = quorumVotes > proposalThreshold
+                ? quorumVotes
+                : proposalThreshold;
+            deal(governanceToken, caller, votingPower);
+
+            // Delegate proposer's votes to itself
+            vm.prank(caller);
+            ERC20Votes(governanceToken).delegate(caller);
+            vm.roll(block.number + 1);
+        }
+
+        uint256 cost = MultichainGovernor(
+            addresses.getAddress("MULTICHAIN_GOVERNOR_PROXY")
+        ).bridgeCostAll();
+
+        vm.deal(address(this), cost);
+        (bool success, bytes memory err) = addresses
+            .getAddress("MULTICHAIN_GOVERNOR_PROXY")
+            .call{value: cost}(execData);
+
+        uint256 proposalId = abi.decode(err, (uint256));
+        require(success, "propose failed");
```

Comment:
I think this check can be moved up and decode `proposalId` only if call is successful